src/price-fetcher.ts

```diff
@@ -156,14 +156,21 @@ class PriceFetcher {
     }
   }
 
-  private updateTokenPrice(coingeckoData, token: Token): void {
+  private async updateTokenPrice(coingeckoData, token: Token): Promise<void> {
     try {
       const price = this.fetchPrice(coingeckoData, token)
       token.price = price;
     } catch (err)  {
       if ((err as HTTPError).code != null) {
-        console.error(`Unable to fetch price data for token ${token.symbol}. ChainID: ${token.chainId}, address ${token.address} Error code is: ${err.code}`);
-        token.noPriceData = true;
+        console.error(`Unable to fetch price data for token ${token.symbol}. ChainID: ${token.chainId}, address ${token.address} Error code is: ${err.code}. Falling back to Balancer SDK`);
+        const coingeckoPriceRepository = new CoingeckoPriceRepository([], token.chainId);
+        const tokenPrice = await coingeckoPriceRepository.find(token.address);
```

Comment:
Just realised we might run into coingecko rate limiting issues here since all the requests will be coming from a single IP. We might need to pay for an account?

Reply:
It's ok at the moment as this price fetcher requests tokens in bulk, one request per 100 tokens per network at the moment, and coingecko rate limits to 50 requests per minute. 

This section of code is falling back to the SDK for price fetching and the reason for that is the SDK has mappings from `aToken -> underlying token` so it can get the prices for wrapped aave tokens in the boosted pool. So this fallback only happens for 3 tokens at the moment. 

In the future we could add bulk price fetching to the SDK and use that as the primary price fetcher.